packages/node_modules/nav-frontend-veileder/src/veileder.tsx

```diff
@@ -42,90 +35,80 @@ interface FrameStyle {
     height?: number;
 };
 
-class Veileder extends React.Component<VeilederProps> { // eslint-disable-line react/prefer-stateless-function
-    public static defaultProps: Partial<VeilederProps> = {
-        type: 'normal'
+export interface NavVeilederSnakkebobleProps {
+    sier: React.ReactNode | React.ReactChild | React.ReactChildren;
+    offset: number;
+};
+
+const NavVeilederSnakkeboble: React.StatelessComponent<NavVeilederSnakkebobleProps> = props => {
+    const { sier, offset } = props;
+
+    const scaleMultiplier = Math.max(0.5, offset / 150);
+
+    const bubbleStyles = {
+        bottom: offset + ((20 * scaleMultiplier) / 2),
+        marginRight: offset / 2
     };
 
+    const arrowStyles = {
+        bottom: Math.ceil(-20 - (((20 * scaleMultiplier) / 2) - 10)),
+        right: ((20 - (20 * scaleMultiplier)) * -1) + (20 * scaleMultiplier),
+        transform: `scale(${scaleMultiplier})`
+    };
+
+    return (
+        <div className="nav-veileder__snakkeboble" style={bubbleStyles}>
+            {sier}
+            <i className="nav-veileder__snakkeboblePil" style={arrowStyles} />
+        </div>
+    );
+};
+
+class NavVeileder extends React.Component<NavVeilederProps> { // eslint-disable-line react/prefer-stateless-function
+    static defaultProps: Partial<NavVeilederProps> = {
+        bredde: 230
+    };
+
+    getChatBubble(veilederBredde) {
+        if (typeof this.props.sier !== 'undefined' && this.props.sier) {
+            return <NavVeilederSnakkeboble sier={this.props.sier} offset={veilederBredde} />
+        }
+    }
+
     render() {
         const {
             children,
-            says,
-            color,
+            sier,
+            farge,
             transparent,
-            width = 230,
+            bredde,
             ...other
         } = this.props;
 
         const domProps = omit(
             { ...other },
-            'center',
-            'nomask'
+            'sentrer',
+            'utenMaske'
         );
 
         const frameStyles: FrameStyle = {};
 
-        if (!transparent && color) {
-            frameStyles.background = color;
+        if (!transparent && farge) {
+            frameStyles.background = farge;
         }
 
-        if (width) {
-            frameStyles.width = width;
-            frameStyles.height = width;
+        if (bredde) {
+            frameStyles.width = bredde;
+            frameStyles.height = bredde;
         }
 
         return (
             <div className={cls(this.props)} {...domProps}>
                 <div className={frameCls(this.props)} style={frameStyles}>{children}</div>
-                {getChatBubble(says, width)}
+                {this.getChatBubble(bredde)}
             </div>
         );
     }
 }
 
-(Veileder as any).propTypes = {
```

Comment:
Hvordan vil dette se ut i guideline-appen om prop-types blir fjernet?

Reply:
Litt buggy, men kanskje akseptabelt inntil videre? Vi bør sikkert uansett legge inn et issue på en litt mer omfattende rewrite av hvordan Guideline-siden funker med docgen for TS osv.

<img width="757" alt="screen shot 2018-02-23 at 08 49 34" src="https://user-images.githubusercontent.com/74510/36583732-2f8ddfda-1877-11e8-976b-79d9763e5fa1.png">

Reply:
Tenker at mange applikasjoner bruker JS her på NAV, så fint å ha propTypes for disse :)